src/raw.rs

```diff
@@ -210,271 +209,19 @@ where
-                // Otherwise, destroy the task right away.
-                Self::destroy(ptr);
-            }
-        }
-    }
-
-    /// Drops a task reference (`Runnable` or `Waker`).
-    ///
-    /// This function will decrement the reference count. If it drops down to zero and the
-    /// associated `Task` handle has been dropped too, then the task gets destroyed.
-    #[inline]
-    unsafe fn drop_ref(ptr: *const ()) {
-        let raw = Self::from_ptr(ptr);
-
-        // Decrement the reference count.
-        let new = (*raw.header).state.fetch_sub(REFERENCE, Ordering::AcqRel) - REFERENCE;
-
-        // If this was the last reference to the task and the `Task` has been dropped too,
-        // then destroy the task.
-        if new & !(REFERENCE - 1) == 0 && new & TASK == 0 {
-            Self::destroy(ptr);
-        }
-    }
-
-    /// Schedules a task for running.
-    ///
-    /// This function doesn't modify the state of the task. It only passes the task reference to
-    /// its schedule function.
-    unsafe fn schedule(ptr: *const (), info: ScheduleInfo) {
-        let raw = Self::from_ptr(ptr);
-
-        // If the schedule function has captured variables, create a temporary waker that prevents
-        // the task from getting deallocated while the function is being invoked.
-        let _waker;
-        if mem::size_of::<S>() > 0 {
-            _waker = Waker::from_raw(Self::clone_waker(ptr));
-        }
-
-        let task = Runnable::from_raw(NonNull::new_unchecked(ptr as *mut ()));
-        (*raw.schedule).schedule(task, info);
-    }
-
-    /// Drops the future inside a task.
-    #[inline]
-    unsafe fn drop_future(ptr: *const ()) {
-        let raw = Self::from_ptr(ptr);
-
-        // We need a safeguard against panics because the destructor can panic.
-        abort_on_panic(|| {
-            raw.future.drop_in_place();
-        })
-    }
-
-    /// Returns a pointer to the output inside a task.
-    unsafe fn get_output(ptr: *const ()) -> *const () {
-        let raw = Self::from_ptr(ptr);
-        raw.output as *const ()
-    }
-
-    /// Cleans up task's resources and deallocates it.
-    ///
-    /// The schedule function will be dropped, and the task will then get deallocated.
-    /// The task must be closed before this function is called.
-    #[inline]
-    unsafe fn destroy(ptr: *const ()) {
-        let raw = Self::from_ptr(ptr);
-        let task_layout = Self::task_layout();
-
-        // We need a safeguard against panics because destructors can panic.
-        abort_on_panic(|| {
-            // Drop the header along with the metadata.
-            (raw.header as *mut Header<M>).drop_in_place();
-
-            // Drop the schedule function.
-            (raw.schedule as *mut S).drop_in_place();
-        });
-
-        // Finally, deallocate the memory reserved by the task.
-        alloc::alloc::dealloc(ptr as *mut u8, task_layout.layout);
+    fn header<'a>(ptr: *const ()) -> &'a Header {
```

Comment:
Mark this as `unsafe`

Reply:
Just went ahead and deleted the function. It's was only used in one location.